### PR TITLE
update comments in segment_reduction_op

### PR DESCRIPTION
**Pull request by jspark1105**

### CODE REVIEW

File: caffe2/operators/segment_reduction_op.cc [hunk 1]
```diff
@@ -19,6 +19,7 @@
 namespace caffe2 {
 
 // registering 4 input gradient with main output
```
Comment (kuttas) [inline caffe2/operators/segment_reduction_op.cc]: NumInputs is 5 here, but the comment says 4?

File: caffe2/operators/segment_reduction_op.h [hunk 2]
```diff
@@ -1615,7 +1615,7 @@ class AbstractLengthsGradientOp : public Operator<Context> {
   }
 
   // Input layout:
-  //   orig_arg1, orig_arg2, ..., orig_argN, SEGMENT_GRADS, SEGMENT_IDS
+  //   orig_arg1, orig_arg2, ..., orig_argN, SEGMENT_GRADS, LENGTHS, INDICES
```
Comment (kuttas) [inline caffe2/operators/segment_reduction_op.h:1618]: is INDICES optional/conditional? Should it be "[INDICES]" in the comment like it is further below in the file?

File: caffe2/operators/segment_reduction_op.h [hunk 3]
```diff
@@ -1731,8 +1731,8 @@ class AbstractLengthsWithMainInputGradientOp : public Operator<Context> {
   }
 
   // Input layout:
-  //   orig_arg1, orig_arg2, ..., orig_argN, DATA_INPUT, SEGMENT_GRADS,
-  //      SEGMENT_LEGNTHS, [INDICES]
+  //   orig_arg1, orig_arg2, ..., orig_argN, SEGMENT_GRADS, LENGTHS,
+  //      DATA_INPUT, [INDICES]
   // orig_argXs represent original op's inputs and will be passed to the reducer
   // directly
   static constexpr int kNumInputs = ReducerGradient::originalInputs().size() +
```
Comment (kuttas) [inline caffe2/operators/segment_reduction_op.h:1738]: Looks like if SparseFused and GradientNeedIndices are both set, there are actually 5 extra inputs?

**Review by kuttas — APPROVED** [review]
Minor questions/nits in the comment but looks good otherwise

File: caffe2/operators/segment_reduction_op.h (hunk 2 above)
Comment (jspark1105) [reply caffe2/operators/segment_reduction_op.h:1618]: INDICES is not optional (required for GPU async version) even though it's not actually used in CPU version.

Comment (jspark1105) [reply caffe2/operators/segment_reduction_op.h:1618]: INDICES is not optional (required for GPU async version) even though it's not used in CPU version.

File: caffe2/operators/segment_reduction_op.h (hunk 3 above)
Comment (jspark1105) [reply caffe2/operators/segment_reduction_op.h:1738]: Strangely, GradientNeedIndices is not used anywhere as far as I can tell. Let me merge this for now and look at this further.

**Pull request merged**